contracts/src/v0.8/functions/dev/1_0_0/FunctionsRouter.sol

```diff
@@ -383,6 +383,18 @@ contract FunctionsRouter is IFunctionsRouter, FunctionsSubscriptions, Pausable,
     uint32 callbackGasLimit,
     address client
   ) private returns (CallbackResult memory) {
```

Comment:
Can we get @se3000 or @RensR to review this? I am not confident enough it my assembly skills to feel comfortable approving.

Reply:
It only moves the check up a bit, assembly logic is almost the same so looks good.

Reply:
I would like to see unit tests cover this logic before merging this